File: packages/types/comment.ts
/**
 * Типы для системы комментариев
 */

export interface Comment {
  id: string;
  userId: string;
  userName: string;
  userAvatar?: string;
  articleId: string;
  articleSlug: string;
  articleTitle: string;
  content: string;
  createdAt: string;
  updatedAt?: string;
  /** ID родительского комментария (для ответов) */
  parentId?: string;
}

/** Корневой комментарий и его ответы для отображения под статьёй */
export interface ArticleCommentThread {
  root: Comment;
  replies: Comment[];
}

export interface CommentSearchParams {
  query?: string;
  limit?: number;
  offset?: number;
}

export interface CommentStats {
  total: number;
  thisWeek: number;
  thisMonth: number;
}

File: apps/api/src/schemas/comment.py
"""
PW-027 | Комментарии: CommentResponse (плоский) + CommentThreadResponse (root + replies).
user_name/avatar и article_slug/title подтягиваются из связей, не хранятся отдельно.
"""

from datetime import datetime

from pydantic import BaseModel


class CommentResponse(BaseModel):
    id: str
    user_id: str
    user_name: str
    user_avatar: str | None = None
    article_id: str
    article_slug: str
    article_title: str
    content: str
    created_at: datetime
    updated_at: datetime | None = None
    parent_id: str | None = None

    model_config = {"from_attributes": True}


class CommentThreadResponse(BaseModel):
    root: CommentResponse
    replies: list[CommentResponse] = []
